pkg/proxy/iptables/proxier.go

```diff
@@ -713,6 +715,9 @@ func (proxier *Proxier) syncProxyRules() {
 		if proxier.masqueradeAll {
 			writeLine(natRules, append(args, "-j", string(kubeMarkMasqChain))...)
 		}
+		if len(proxier.clusterCIDR) > 0 {
```

Comment:
Yeah, you're right.  It's unfortunate, but any simpler solution is either too broad or costs the same number of rules.